src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -948,6 +953,100 @@ class spell_warr_vigilance_trigger : public SpellScriptLoader
         }
 };
 
+// Heroic leap - 6544
+class spell_warr_heroic_leap : public SpellScriptLoader
+{
+public:
+    spell_warr_heroic_leap() : SpellScriptLoader("spell_warr_heroic_leap") { }
+
+    class spell_warr_heroic_leap_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_warr_heroic_leap_SpellScript);
+
+        bool Validate(SpellInfo const* /*spellInfo*/) override
+        {
+            if (!sSpellMgr->GetSpellInfo(SPELL_WARRIOR_HEROIC_LEAP_JUMP))
+                return false;
+            return true;
+        }
+
+        SpellCastResult CheckElevation()
+        {
+            if (WorldLocation const* dest = GetExplTargetDest())
+            {
+                if (dest->GetPositionZ() > GetCaster()->GetPositionZ() + 4.0f)
```

Comment:
![wowscrnshot_053115_152404](https://cloud.githubusercontent.com/assets/799249/7902073/b3999a58-07a9-11e5-821e-f34fc0e7ed0e.jpg)
![wowscrnshot_053115_152411](https://cloud.githubusercontent.com/assets/799249/7902074/b3a06234-07a9-11e5-8c55-ce316f3937a4.jpg)

As far as I know it doesn't use the pathfinding system, from playing around with it on retail it suggests that it uses height comparsion and the z value is around 4yards.